runtime/test/type-checker-tests.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import {assert} from './chai-web.js';
+
+import Schema from '../schema.js';
+import Type from '../type.js';
+import TypeChecker from '../recipe/type-checker.js';
+import TypeVariable from '../type-variable.js';
+
+
+describe('TypeChecker', () => {
+  it('resolves a trio of in [~a], out [~a], in [Product]', async () => {
+    let a = Type.newVariable(new TypeVariable('a')).setViewOf();
+    let b = Type.newVariable(new TypeVariable('a')).setViewOf();
```

Comment:
Clearer to call this ~b since it's not merged with the other ~a.

Reply:
Done